lib/mix/test/mix/tasks/help_test.exs

```diff
@@ -70,6 +142,13 @@ defmodule Mix.Tasks.HelpTest do
       assert_received {:mix_shell, :info, ["mix deps" <> _]}
       assert_received {:mix_shell, :info, ["mix deps.clean" <> _]}
     end)
+
+    Mix.Project.push(Aliases)
+
+    in_tmp(context.test, fn ->
+      Mix.Tasks.Help.run(["--search", "h"])
+      assert_received {:mix_shell, :info, ["mix h" <> _]}
```

Comment:
Unfortunately, this might also give false positives with `mix help` itself, `mix hex` or any `mix hex.*`.

Reply:
I would either include a space, or match on the message live above:

```
assert message =~ ~r/# Alias defined in mix.exs/
```